openff/interchange/_tests/unit_tests/smirnoff/test_nonbonded.py

```diff
@@ -135,6 +135,347 @@ def test_toolkit_am1bcc_uses_elf10_if_oe_is_available(self, sage, hexane_diol):
+        from openff.toolkit.utils.toolkit_registry import ToolkitRegistry, toolkit_registry_manager
+
+        from openff.interchange import Interchange
+
+        # Mock the toolkit registry to not have NAGL
+        # RDKit is needed for SMARTS matching.
+        with toolkit_registry_manager(ToolkitRegistry(toolkit_precedence=[RDKitToolkitWrapper])):
+            ff = ForceField("openff-2.1.0.offxml")
+            ff.get_parameter_handler(
+                "NAGLCharges",
+                {
+                    "model_file": "openff-gnn-am1bcc-0.1.0-rc.3.pt",
+                    "version": "0.3",
+                },
+            )
+
+            with pytest.raises(MissingPackageError, match="NAGL software isn't present"):
+                Interchange.from_smirnoff(force_field=ff, topology=hexane_diol.to_topology())
+
+    def test_nagl_charges_invalid_model_file(self, hexane_diol):
+        """Test error handling for invalid model file paths."""
+        from openff.toolkit import ForceField
+
+        from openff.interchange import Interchange
+
+        ff = ForceField("openff-2.1.0.offxml")
+        ff.get_parameter_handler(
+            "NAGLCharges",
+            {
+                "model_file": "nonexistent_model.pt",
+                "version": "0.3",
+            },
+        )
+        with pytest.raises(ValueError, match="No registered toolkits can provide the capability"):
+            Interchange.from_smirnoff(force_field=ff, topology=hexane_diol.to_topology())
+
+    def test_nagl_charges_empty_model_file(self, hexane_diol):
+        """Test error handling for empty model file parameter."""
+        from openff.toolkit.typing.engines.smirnoff import ForceField
+
+        from openff.interchange import Interchange
+
+        ff = ForceField("openff-2.1.0.offxml")
+        ff.get_parameter_handler(
+            "NAGLCharges",
+            {
+                "model_file": "",
+                "version": "0.3",
+            },
+        )
+        with pytest.raises(ValueError, match="No registered toolkits can provide the capability"):
+            Interchange.from_smirnoff(force_field=ff, topology=hexane_diol.to_topology())
+
+    def test_nagl_charges_none_model_file(self, hexane_diol):
+        """Test error handling for None model file parameter."""
+        from openff.toolkit.typing.engines.smirnoff import ForceField
+
+        from openff.interchange import Interchange
+
+        ff = ForceField("openff-2.1.0.offxml")
+        ff.get_parameter_handler(
+            "NAGLCharges",
+            {
+                "model_file": None,
+                "version": "0.3",
+            },
+        )
+        with pytest.raises(ValueError, match="No registered toolkits can provide the capability"):
+            Interchange.from_smirnoff(force_field=ff, topology=hexane_diol.to_topology())
+
+
+class TestNAGLChargesPrecedence:
+    """Test NAGLCharges precedence over other charge handlers."""
```

Comment:
```suggestion
    """Test NAGLCharges precedence in the hierarchy of charge assignment methods."""
```

When I first read this I thought it meant NAGL was always at the top of the hierarchy